extra_data/tests/test_reader_mockdata.py

```diff
@@ -742,22 +742,37 @@ def test_train_timestamps_nat(mock_fxe_control_data):
 def test_union(mock_fxe_raw_run):
     run = RunDirectory(mock_fxe_raw_run)
 
-    sel1 = run.select('SPB_XTD9_XGM/DOOCS/MAIN', 'beamPosition.ixPos')
-    sel2 = run.select('SPB_XTD9_XGM/DOOCS/MAIN', 'beamPosition.iyPos')
+    xgm = "SPB_XTD9_XGM/DOOCS/MAIN"
+    camera = "FXE_XAD_GEC/CAM/CAMERA"
+
+    # Test union of different sources
+    sel1 = run.select(xgm, 'beamPosition.ixPos')
+    sel2 = run.select(xgm, 'beamPosition.iyPos')
     joined = sel1.union(sel2)
-    assert joined.control_sources == {'SPB_XTD9_XGM/DOOCS/MAIN'}
+    assert joined.control_sources == { xgm }
     assert joined.selection == {
-        'SPB_XTD9_XGM/DOOCS/MAIN': {
+        xgm: {
             'beamPosition.ixPos.value',
             'beamPosition.iyPos.value',
         }
     }
 
+    # Test union of different train selections
     sel1 = run.select_trains(by_id[10200:10220])
     sel2 = run.select_trains(by_index[:10])
     joined = sel1.union(sel2)
     assert joined.train_ids == list(range(10000, 10010)) + list(range(10200, 10220))
 
+    # Test union of different sources in different train selections
+    sel1 = run.select(xgm).select_trains(by_index[:5])
+    sel2 = run.select(camera).select_trains(by_index[-5:])
+    joined = sel1.union(sel2)
+    expected_tids = run.train_ids[:5] + run.train_ids[-5:]
+
+    assert joined.train_ids == expected_tids
+    assert joined[xgm].train_ids == expected_tids
+    assert joined[camera].train_ids == expected_tids
```

Comment:
Hmmm, I'll have to think about this one. Intuitively, it feels more correct for union-ing them to keep the separate train IDs which are selected for each, and not re-select them. But it also feels desirable that `run.train_ids == run[src].train_ids`, and this would be more consistent with going from source to key (KeyData objects are created on demand, rather than always existing).

Reply:
Added some docs in b65658e.

Reply:
(bump :) )